v3/generator/servers/servers.go

```diff
@@ -0,0 +1,319 @@
+// Note: the Exoscale API client will chain additional middleware
+// (http.RoundTripper) on the HTTP client internally, which can alter the HTTP
+// requests and responses. If you don't want any other middleware than the ones
+// currently set to your HTTP client, you should duplicate it and pass a copy
+// instead.
+func Client{{.APIName}}OptWithHTTPClient(v *http.Client) Client{{.APIName}}Opt {
+	return func(c *Client{{.APIName}}) error {
+		c.httpClient = v
+
+		return nil
+	}
+}
+
+func NewClient{{.APIName}}(apiKey, apiSecret string, opts ...Client{{.APIName}}Opt) (Client, error) {
+  if apiKey == "" || apiSecret == "" {
+	return nil, fmt.Errorf("missing or incomplete API credentials")
+  }
+
+  client := &Client{{.APIName}}{
+	apiKey:				apiKey,
+	apiSecret:			apiSecret,
+	serverURL:			"{{.ServerURL}}",
+    rawServerURL:		"{{.RawServerURL}}",
+	httpClient:			http.DefaultClient,
+	ctx:				context.Background(),
+	pollingInterval:	pollingInterval,
+  }
+
+  for _, opt := range opts {
+	if err := opt(client); err != nil {
+		return nil, fmt.Errorf("client configuration error: %s", err)
+	}
+  }
+
+  security, err := api.NewSecurityProvider(apiKey, apiSecret)
+  if err != nil {
+	  return nil, fmt.Errorf("unable to initialize API security provider: %w", err)
+  }
+
+  // Tracing must be performed before API error handling in the middleware chain,
+  // otherwise the response won't be dumped in case of an API error.
+  if client.trace {
+    client.httpClient.Transport = api.NewTraceMiddleware(client.httpClient.Transport)
+  }
+
+  client.httpClient.Transport = api.NewAPIErrorHandlerMiddleware(client.httpClient.Transport)
+
+  return client.WithRequestMiddleware(security.Intercept), nil
+}
+
+func (c *Client{{.APIName}}) WithZone(z {{ .APIName }}Zone) Client {
+  return &Client{{.APIName}}{
+	serverURL:			strings.Replace(c.rawServerURL, "{zone}", string(z), 1),
+    rawServerURL:  		c.rawServerURL,
+	httpClient:			c.httpClient,
+	ctx:				c.ctx,
+	middlewares:  		c.middlewares,
+	pollingInterval:	pollingInterval,
+  }
+}
+
+func (c *Client{{.APIName}}) WithContext(ctx context.Context) Client {
+	return &Client{{.APIName}}{
+	  serverURL:		c.serverURL,
+	  rawServerURL:		c.rawServerURL,
+	  httpClient:		c.httpClient,
+	  ctx:				ctx,
+	  middlewares:  	c.middlewares,
+	  pollingInterval:	pollingInterval,
+	}
+  }
+
+  func (c *Client{{.APIName}}) WithHttpClient(client *http.Client) Client {
+	return &Client{{.APIName}}{
+	  serverURL:		c.serverURL,
+	  rawServerURL:		c.rawServerURL,
+	  httpClient:		client,
+	  ctx:				c.ctx,
+	  middlewares:  	c.middlewares,
+	  pollingInterval:	pollingInterval,
+	}
+  }
+
+  func (c *Client{{.APIName}}) WithRequestMiddleware(f RequestMiddlewareFn) Client {
+	return &Client{{.APIName}}{
+		serverURL:    		c.serverURL,
+		rawServerURL: 		c.rawServerURL,
+		httpClient:   		c.httpClient,
+		ctx:          		c.ctx,
+		middlewares:  		append(c.middlewares, f),
+		pollingInterval:	pollingInterval,
+	}
+  }
```

Comment:
I don't get why the original c.pollingInterval isn't copied here

Reply:
You spotted a typo here thanks a lot :)